src/kerberos.c

```diff
@@ -585,13 +585,146 @@ static PyObject *authGSSServerTargetName(PyObject *self, PyObject *args) {
     return Py_BuildValue("s", state->targetname);
 }
 
+static PyObject* authGSSWinRMEncryptMessage(PyObject* self, PyObject* args)
+{
+    char *input = NULL;
+    char *header = NULL;
+    int header_len = 0;
+    char *enc_output = NULL;
+    int enc_output_len = 0;
+    PyObject *pystate = NULL;
+    gss_client_state *state = NULL;
+    int result = 0;
+    PyObject *pyresult = NULL;
+
+    // NB: use et so we get a copy of the string (since gss_wrap_iov mutates it), and so we're certain it's always
+    // a UTF8 byte string
+    if (! PyArg_ParseTuple(args, "Oet", &pystate, "UTF-8", &input)) {
+        pyresult = NULL;
```

Comment:
Any reason why your don't just say `return NULL` like the other functions, IIRC Python will throw an error if you run `PyErr_SetString()` before it.